worlds/oot_soh/location_access/dungeons/fire_temple.py

```diff
@@ -2,8 +2,518 @@
+            (is_adult(bundle) and
+             (can_do_trick(Tricks.FIRE_BOSS_DOOR_JUMP, bundle) or 
+              has_item(LocalEvents.FIRE_TEMPLE_FIRE_MAZE_UPPER_PLATFORM_HIT, bundle) or 
+              can_use(Items.HOVER_BOOTS, bundle))))
+    ])
+    
+    ## Fire Temple Loop Enemies
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_LOOP_ENEMIES, world, [
+        (Regions.FIRE_TEMPLE_FIRST_ROOM, lambda bundle: 
+            (small_keys(Items.FIRE_TEMPLE_SMALL_KEY, 8, bundle) or not is_fire_loop_locked(bundle))),
+        (Regions.FIRE_TEMPLE_LOOP_TILES, lambda bundle:
+            (can_kill_enemy(bundle, Enemies.TORCH_SLUG) and
+             can_kill_enemy(bundle, Enemies.FIRE_KEESE)))
+    ])
+    
+    ## Fire Temple Loop Tiles
+    # Locations
+    add_locations(Regions.FIRE_TEMPLE_LOOP_TILES, world, [
+        (Locations.FIRE_TEMPLE_GS_BOSS_KEY_LOOP, lambda bundle: can_attack(bundle))
+    ])
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_LOOP_TILES, world, [
+        (Regions.FIRE_TEMPLE_LOOP_ENEMIES, lambda bundle: True),
+        (Regions.FIRE_TEMPLE_LOOP_FLARE_DANCER, lambda bundle: True)
+    ])
+    
+    ## Fire Temple Loop Flare Dancer
+    # Locations
+    add_locations(Regions.FIRE_TEMPLE_LOOP_FLARE_DANCER, world, [
+        (Locations.FIRE_TEMPLE_FLARE_DANCER_CHEST, lambda bundle:
+            (has_explosives(bundle) or
+             can_use(Items.MEGATON_HAMMER, bundle)) and
+            is_adult(bundle))
+    ])
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_LOOP_FLARE_DANCER, world, [
+        (Regions.FIRE_TEMPLE_LOOP_TILES, lambda bundle: True),
+        (Regions.FIRE_TEMPLE_LOOP_HAMMER_SWITCH, lambda bundle:
+            (can_kill_enemy(bundle, Enemies.FLARE_DANCER)))
+    ])
+    
+    ## Fire Temple Loop Hammer Switch
+    # Events
+    add_events(Regions.FIRE_TEMPLE_LOOP_HAMMER_SWITCH, world, [
+        (EventLocations.FIRE_TEMPLE_LOOP_HAMMER_SWITCH_ROOM_SWITCH, LocalEvents.FIRE_TEMPLE_LOOP_HAMMER_SWITCH_HIT, lambda bundle:
+            (can_use(Items.MEGATON_HAMMER, bundle)))
+    ])
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_LOOP_HAMMER_SWITCH, world, [
+        (Regions.FIRE_TEMPLE_LOOP_FLARE_DANCER, lambda bundle: True),
+        (Regions.FIRE_TEMPLE_LOOP_GORON_ROOM, lambda bundle:
+            (has_item(LocalEvents.FIRE_TEMPLE_LOOP_HAMMER_SWITCH_HIT, bundle)))
+    ])
+    
+    ## Fire Temple Loop Goron Room
+    # Locations
+    add_locations(Regions.FIRE_TEMPLE_LOOP_GORON_ROOM, world, [
+        (Locations.FIRE_TEMPLE_BOSS_KEY_CHEST, lambda bundle: True)
+    ])
+    # Connections
+    connect_regions(Regions.FIRE_TEMPLE_LOOP_GORON_ROOM, world, [
+        (Regions.FIRE_TEMPLE_LOOP_HAMMER_SWITCH, lambda bundle:
+            (has_item(LocalEvents.FIRE_TEMPLE_LOOP_HAMMER_SWITCH_HIT, bundle))),
+        (Regions.FIRE_TEMPLE_LOOP_EXIT, lambda bundle:
+            (has_item(LocalEvents.FIRE_TEMPLE_LOOP_HAMMER_SWITCH_HIT, bundle)))
```

Comment:
I will just note this down once because it is a nitpick, but these parenthesis are unnecessary.
```suggestion
            has_item(LocalEvents.FIRE_TEMPLE_LOOP_HAMMER_SWITCH_HIT, bundle))
```